src/pt/toonbr/src/eu/kanade/tachiyomi/extension/pt/toonbr/ToonBr.kt

```diff
@@ -0,0 +1,196 @@
+package eu.kanade.tachiyomi.extension.pt.toonbr
+
+import androidx.preference.EditTextPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.lib.cookieinterceptor.CookieInterceptor
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import keiyoushi.utils.getPreferencesLazy
+import keiyoushi.utils.parseAs
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ToonBr : HttpSource(), ConfigurableSource {
+
+    private val preferences by getPreferencesLazy()
+
+    override val name = "ToonBr"
+    override val baseUrl = "https://beta.toonbr.com"
+    override val lang = "pt-BR"
+    override val supportsLatest = true
+
+    override val client by lazy {
+        val token = getToken()
+        network.cloudflareClient.newBuilder()
+            .rateLimit(2)
+            .apply {
+                if (token.isNotEmpty()) {
+                    addInterceptor(CookieInterceptor(API_HOST, "token" to token))
+                }
+            }
+            .build()
+    }
+
+    private val apiUrl = "https://api.toonbr.com"
+    private val cdnUrl = "https://cdn2.toonbr.com"
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.ROOT)
+    }
+
+    private fun getToken(): String {
+        val email = preferences.getString(PREF_EMAIL, "") ?: ""
+        val password = preferences.getString(PREF_PASSWORD, "") ?: ""
+        if (email.isEmpty() || password.isEmpty()) {
+            return ""
+        }
+        return runCatching { login(email, password) }.getOrDefault("")
+    }
+
+    // ===== Popular =====
+    override fun popularMangaRequest(page: Int): Request {
```

Comment:
Make sure you override `getMangaUrl` and `getChapterUrl` so the user isn't directly exposed to the API.